File: AspNetMvcUnleashed/0672329980_source/Chapter15/VB/MvcApplication1/MvcApplication1/Controllers/NewsController.vb
Public Class NewsController
    Inherits Controller

    Public Function Index() As ActionResult
        Return View()
    End Function

    Public Function Refresh() As ActionResult
        Dim newsPartial = ""
        Dim rnd = New Random()
        Select Case rnd.Next(2)
            Case 0
                newsPartial = "News/News1"
            Case 1
                newsPartial = "News/News2"
        End Select

        Return PartialView(newsPartial)
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter15/CS/MvcApplication1/MvcApplication1/Controllers/NewsController.cs
using System;
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class NewsController : Controller
    {

        public ActionResult Index()
        {
            return View();
        }

        public ActionResult Refresh()
        {
            var partial = "";
            var rnd = new Random();
            switch (rnd.Next(2))
            {
                case 0:
                    partial = "News/News1";
                    break;
                case 1:
                    partial = "News/News2";
                    break;
            }

            return PartialView(partial);
        }

    }
}
